domain/network/service/space.go

```diff
@@ -61,6 +62,43 @@ func NewSpaceService(st State, logger Logger) *SpaceService {
 	}
 }
 
+// AddSpace creates and returns a new space.
+func (s *SpaceService) AddSpace(ctx context.Context, uuid utils.UUID, name string, providerID network.Id, subnetIDs []string) (*network.SpaceInfo, error) {
+	if !names.IsValidSpace(name) {
+		return nil, errors.NotValidf("space name %q for space with uuid %q", name, uuid.String())
+	}
+
+	if err := s.st.AddSpace(ctx, uuid, name, providerID, subnetIDs); err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	return s.st.GetSpace(ctx, uuid.String())
```

Comment:
I wonder if it's just better to return the space from `AddSpace`, otherwise it's two database transactions.

Reply:
As discussed offline, I removed this call so now the `AddSpace()` method will return the newly created space UUID only.